agent/lib_monolog.c

```diff
@@ -298,44 +243,65 @@ static char* nr_monolog_get_context(const size_t argc,
     goto return_context;
   }
 
-  context = nr_monolog_fmt_context(context, Z_ARRVAL_P(context_arg) TSRMLS_CC);
-
 return_context:
-  nr_php_arg_release(&context_arg);
-  return context;
+  return context_arg;
+}
+
+/*
+ * Purpose : Convert $context array of Monolog\Logger::addRecord to
+ * attributes
+ *
+ * Params  : zval* for context array from Monolog
+ *
+ * Returns : nr_attributes representation of $context on success
+ *           NULL otherwise
+ *
+ */
+nr_attributes_t* nr_monolog_convert_context_data_to_attributes(
+    zval* context_data TSRMLS_DC) {
+  zend_string* key;
+  zval* val;
+
+  nr_attributes_t* attributes = NULL;
+
+  if (NULL == context_data || !nr_php_is_zval_valid_array(context_data)) {
+    return NULL;
+  }
+
+  attributes = nr_attributes_create(NRPRG(txn)->attribute_config);
+
+  ZEND_HASH_FOREACH_STR_KEY_VAL(Z_ARR_P(context_data), key, val) {
+    if (NULL == key) {
+      continue;
+    }
+
+    nrobj_t* obj = nr_monolog_context_data_zval_to_attribute_obj(val);
+
+    if (NULL != obj) {
+      nr_attributes_user_add(attributes, NR_ATTRIBUTE_DESTINATION_LOG,
+                                  ZSTR_VAL(key), obj);
+      nro_delete(obj);
+    }
+  }
+  ZEND_HASH_FOREACH_END();
+
+  return attributes;
 }
-#endif
 
 /*
  * Purpose : Combine $message and $context arguments of
```

Comment:
```suggestion
 * Purpose : Convert $message argument of
```

Reply:
I removed this function so I believe this suggestion is no longer applicable.